tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs

```diff
@@ -161,61 +197,205 @@ internal Configuration BuildXmlConfiguration(IEnumerable<Type> cmdletTypList)
             return configuration;
         }
 
+        internal class GroupByInfo {
+            public  string Label { get; set; }
+            public  string PropertyName { get; set; }
+            public  string ScriptBlock { get; set; }
+            public ViewControl TargetView { get; set; }
+        }
+
         internal void HandleMemberInfo<T>(
             IEnumerable<T> memberInfoList,
             IList<TableColumnHeader> tableHeaders, 
             IList<TableColumnItem> tableColumnItems, 
-            IList<ListItem> listItems) 
+            IList<ListItem> listItems,
+            ref GroupByInfo groupByInfo,
+            IList<string> warnings) 
             where T: MemberInfo
         {
-
             var memeber = typeof(T) == typeof(PropertyInfo)
                 ? "property"
                 : "field";
 
             var memberInfoListFiltered = memberInfoList
                 .Where(p => {
                     if (!_onlyMarkedProperties) return true;
-                    var ps1XmlAttribute = (Ps1XmlAttribute)Attribute.GetCustomAttribute(p, typeof(Ps1XmlAttribute));
-                    return ps1XmlAttribute != null;
+                    var ps1XmlAttributes = Attribute.GetCustomAttributes(p, typeof(Ps1XmlAttribute));
+                    return ps1XmlAttributes.Length > 0;
                 })
-                .OrderBy(p => p.Name)
                 .ToList();
 
             if (memberInfoListFiltered.Count == 0) return;
 
+            var columnHeadersWithPosition = new List<Tuple<TableColumnHeader, uint>>();
+            var columnItemsWithPosition = new List<Tuple<TableColumnItem, uint>>();
+            var listItemsWithPosition = new List<Tuple<ListItem, uint>>();
+
+            var dupTrackerGroupBy = new List<Tuple<string, string, string>>();
+            var dupTrackerPosition = new List<Tuple<string, string, string, uint>>();
+
             foreach (var memberInfo in memberInfoListFiltered)
```

Comment:
It'll be great if you can add a comment here that explains whats happening in the for loop and how we track duplicates for both position and group by.